js/tabRestore.js

```diff
@@ -0,0 +1,48 @@
+function TabStack() {
+  this.depth = 20
+  this.stack = []
+}
+
+TabStack.prototype.push = function (tabId) {
+  var closedTab = tabs.get(tabId)
+
+  // Do not store private tabs or blank tabs
+  if (closedTab.private
+    || closedTab.url === 'about:blank'
+    || closedTab.url === '') {
+    return
+  }
+
+  if (this.stack.length >= this.depth) {
+    this.stack.shift()
+  }
+
+  this.stack.push(closedTab)
+}
+
+TabStack.prototype.restore = function () {
+  if (this.stack.length === 0) {
+    return
+  }
+
+  lastTab = window.currentTask.tabs[window.tabs.length - 1]
+
+  // Open the tab in the last slot
```

Comment:
Opening tabs next to the current one seems better, since it matches what happens when you open a new tab. Is there a benefit to opening it at the end?